astroid/nodes/node_classes.py

```diff
@@ -635,6 +638,19 @@ def __init__(
 
         super().__init__(lineno=lineno, col_offset=col_offset, parent=parent)
 
+    assigned_stmts: Callable[
+        [
+            "AssignName",
+            AssignedStmtsPossibleNode,
+            Optional[InferenceContext],
+            Optional[typing.List[int]],
+        ],
+        Any,
+    ]
```

Comment:
The type is almost the same for every node. We could try to use a generic type alias.
https://mypy.readthedocs.io/en/latest/generics.html#generic-type-aliases

If PEP 673 gets accepted (and eventually implemented) we could even use `Self` for the node type. https://www.python.org/dev/peps/pep-0673/